File: Container with Most Water.py
height = [1, 1]
i, j = 0, len(height) - 1
maX = -1 * float("inf")
while i <= j:
    if height[i] < height[j]:
        maX = max(maX, height[i] * (j - i))
        i += 1
    elif height[i] > height[j]:
        maX = max(maX, height[j] * (j - i))
        j -= 1
    else:
        maX = max(maX, height[j] * (j - i))
        j -= 1
        i += 1
print(maX)

File: Container With Most Water.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    static int maxArea(const vector<int> &height)
    {
        const int n = height.size();
        int j = n - 1;
        int i = 0;
        int Max = 0;
        int temp = 0;
        while (i < j)
        {
            if (height[j] < height[i])
            {
                temp = (j - i) * (height[j]);
                j--;
            }
            else
            {
                temp = (j - i) * (height[i]);
                i++;
            }
            Max = max(Max, temp);
        }
        return Max;
    }
};

int main(int argc, const char **argv)
{
    const vector<int> nums = {1, 8, 6, 2, 5, 4, 8, 3, 7};
    cout << Solution::maxArea(nums);
}
